circleci: failed tests: TestEagerReplication

The following test appears to have failed:

[#19561](https://circleci.com/gh/cockroachdb/cockroach/19561):

```
I160623 21:30:11.082571 storage/replica.go:1786  applying command with forced error: storage/replica.go:1707: no-op on empty Raft entry
E160623 21:30:11.082804 storage/replica.go:1797  [node=1,store=1,range=4] error executing raft command: storage/replica.go:1707: no-op on empty Raft entry
I160623 21:30:11.082924 storage/replica_command.go:1575  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900.000123ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00.000000123 +0000 UTC]
I160623 21:30:11.083793 storage/replica_command.go:2126  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160623 21:30:11.092415 storage/engine/rocksdb.go:185  closing in-memory rocksdb instance
--- FAIL: TestEagerReplication (0.11s)
	replicate_test.go:45: expected 5 replicas in purgatory, but found 4
=== RUN   Example_rebalancing
W160623 21:30:11.092700 gossip/gossip.go:898  not connected to cluster; use --join to specify a connected node
I160623 21:30:11.193209 stopper.go:352  draining; tasks left:
95     gossip/infostore.go:288
I160623 21:30:11.193290 stopper.go:352  draining; tasks left:
94     gossip/infostore.go:288
I160623 21:30:11.193330 stopper.go:352  draining; tasks left:
93     gossip/infostore.go:288
I160623 21:30:11.193369 stopper.go:352  draining; tasks left:
--
I160623 21:30:11.197399 stopper.go:352  draining; tasks left:
2      gossip/infostore.go:288
I160623 21:30:11.197462 stopper.go:352  draining; tasks left:
1      gossip/infostore.go:288
--- PASS: Example_rebalancing (0.10s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	45.091s
=== RUN   TestBatchBasics
I160623 21:29:21.966593 storage/engine/rocksdb.go:142  opening in memory rocksdb instance
I160623 21:29:21.968966 storage/engine/rocksdb.go:185  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchRepr
I160623 21:29:21.976813 storage/engine/rocksdb.go:142  opening in memory rocksdb instance
I160623 21:29:21.978381 storage/engine/rocksdb.go:185  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.02s)
=== RUN   TestBatchGet
I160623 21:29:21.990559 storage/engine/rocksdb.go:142  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replicate_test.go
+++ b/storage/replicate_test.go
@@ -20,7 +20,9 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/leaktest"
+	"github.com/pkg/errors"
 )
 
 func TestEagerReplication(t *testing.T) {
@@ -36,12 +38,19 @@ func TestEagerReplication(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// After the initial splits have been performed, all of the resulting ranges
-	// should be present in replicate queue purgatory (because we only have a
-	// single store in the test and thus replication cannot succeed).
-	expected := server.ExpectedInitialRangeCount()
-	if n := store.ReplicateQueuePurgatoryLength(); expected != n {
-		t.Fatalf("expected %d replicas in purgatory, but found %d",
-			expected, n)
-	}
+	// WaitForInitialSplits will return as soon as the meta2 span contains the
+	// expected number of descriptors. But the addition of replicas to the
+	// replicateQueue after a split occurs happens after the update of the
+	// descriptors in meta2 leaving a tiny window of time in which the newly
+	// split replica will not have been added to purgatory. Thus we loop.
+	util.SucceedsSoon(t, func() error {
+		// After the initial splits have been performed, all of the resulting ranges
+		// should be present in replicate queue purgatory (because we only have a
+		// single store in the test and thus replication cannot succeed).
+		expected := server.ExpectedInitialRangeCount()
+		if n := store.ReplicateQueuePurgatoryLength(); expected != n {
+			return errors.Errorf("expected %d replicas in purgatory, but found %d", expected, n)
+		}
+		return nil
+	})
 }

